app/controllers/products_controller.rb

```diff
@@ -0,0 +1,51 @@
+class ProductsController < ApplicationController
+
+  def index
+    @categories = Category.order(name: :asc)
+    @merchants = User.joins(:products).uniq.sort
+
+    scope = Product.where(retired: false).order(name: :asc)
```

Comment:
Oooooo Scopes!

Since you might want to use this elsewhere, this might be good to put as a scope in the model.

http://guides.rubyonrails.org/active_record_querying.html#scopes